tests/RegressionTests.hs

```diff
@@ -280,7 +280,7 @@ externalProgramTests = testGroup "ExternalProgram-RegressionTests"
     , mkParseTest "topLevelConsts_sics" sicsOptions
     -- TODO: Enable with a typed array representation.
     -- , mkParseTest "metrics" defaultOptions
-    , mkParseTest "scanlPush" defaultOptions
+    -- , mkParseTest "scanlPush" defaultOptions
```

Comment:
Well, it's the same kind of problem, so I let the comment stand for both lines.

Reply:
Don't we effectively get the same knowledge as a typed array representation would give us with this patch?

Reply:
Nope. The generated C-program still uses 'struct array*' for arrays. Of course, we could do full type inference for C to recover the types, but I find this an overkill since the next step will at  least give us more type information.

Reply:
I agree, if we'll get typed arrays in the next step and that fixes this issue, this test can be disabled for now. I think the rest I mentioned has been fixed except using concatMap instead of concat $ map. (HLint points out a bunch of these small things automatically.)

Reply:
Ouch, missed that one.